crypto/x509/x_x509.c

```diff
@@ -60,9 +60,12 @@ static int x509_cb(int operation, ASN1_VALUE **pval, const ASN1_ITEM *it,
     case ASN1_OP_NEW_POST:
         ret->ex_cached = 0;
         ret->ex_kusage = 0;
+        if (!CRYPTO_THREAD_write_lock(ret->lock))
```

Comment:
Why is the lock just around these 2? What determines this..
It seems like a few others are only set inside ossl_x509v3_cache_extensions which is all inside a lock.

Reply:
These are the only two that are locked elsewhere, I could must have missed some.  Sledgehammer approach coming up :)

Reply:
Have a look at ossl_x509v3_cache_extensions()..

Reply:
I moved all of these into the lock.

Reply:
Isn't this whole thing just a false positive? I mean this is called only when newly creating an X509 object, is there any point in locking it?

Reply:
It's using a X509 structure that's passed in, one which has a lock already allocated.  It's not clear if it is a false positive or not.

Reply:
It is quite clear from what asn1_item_embed_new() does. The lock was initialized right before this callback call so I do not see what would it be protecting against.

Reply:
I've reverted this file and tagged the two issues as false positivies.